src/collision.cpp

```diff
@@ -589,10 +593,10 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 				box.MaxEdge += *pos_f;
 			}
 			if (std::fabs(cbox.MaxEdge.Y - box.MinEdge.Y) < 0.05f) {
+				// This is ugly, hacky and only required if `box_info.is_step_up == true`
```

Comment:
`standing_on_object` is also exposed to the Lua API as a moveresult field

Reply:
Yes. I think you misunderstood the intend of this comment. `step_up` is a special case where this should indeed be set to `true`. However, the collision box distance checks would not be necessary. The collision code above already detects that case reliably.